src/main/java/com/minecolonies/coremod/entity/ai/citizen/student/EntityAIStudy.java

```diff
@@ -136,12 +136,15 @@ private IAIState study()
                 final int bSlot = InventoryUtils.findFirstSlotInProviderWith(building, studyItem.getItem());
                 if (bSlot > -1)
                 {
+                    //Need to move this here. Since worker's previous position was a bookshelf, he'll always walk to the building and then return getState, never actually grabbing the item.
+                    //This causes him to loop back and forth getting the free xp on line 131 and never hitting the delay on 173
+                    takeItemStackFromProvider(building, bSlot); 
                     if (walkToBuilding())
```

Comment:
WalkToBuilding should return true until it arrived at the building and then take the stack on arrival only. The current code will get it from the building even before they started to walk

Reply:
We should probably redirect this to its own state. We already have code for pickup stuff. So instead just set the pickup predicate and put them into pickup state.

Reply:
That makes sense, I guess walking is it's own state? When it finishes the walking what state does it default into? Observation says it goes back to STUDY because it never actually grabs the item and he just walks back to the bookshelf for more sweet sweet xp :D